### ISSUE TITLE
Very long time to show

### ISSUE THREAD

**Issue by hrace009** [bonecms/laravel-captcha#55]
Captcha show with very long time, i don't know why, but this is using localhost with xampp and php 8.1
![image](https://user-images.githubusercontent.com/7691118/163419505-6471f1d1-0bcf-49c6-b933-5ff2444bacf4.png)

**Comment by mishinroman94** [comment]
I have the same problem. Works fine on php 8.0

**Comment by hrace009** [comment]
> 

yeah, last time i use php7.4 it working fine, but when i switch to php8.1 the code show very long time

**Comment by some0ne2** [comment]
Can confirm this, switched from working PHP7.4 to PHP8.1 (tried laravel8 and laravel9), same effect. as soon as I switch PHP to 8.1 the whole page speed is ruined.

Not usable in production.
@igoshev any idea where things go wrong / can you take a look into?

**Comment by hrace009** [comment]
@igoshev please make a fix for this

**Comment by hrace009** [comment]
i don't see any respond from @igoshev

**Comment by hrace009** [comment]
seems we must make our own fix for this package

**Comment by some0ne2** [comment]
@tm1000 any idea maybe since you pushed the commit for laravel9?

**Comment by some0ne2** [comment]
unusable for PHP8.1

**Comment by tm1000** [comment]
@some0ne2 if you look at my commit you can see that I only bumped the composer file

Regardless my company has five services that use this on php 8.1 for the last two-three months and we've experienced no issues

**Comment by some0ne2** [comment]
@tm1000 Thanks for information, what laravel version and what PHP version exactly are you using?

**Comment by tm1000** [comment]
Laravel 9 and php 8.1.7

**Comment by tm1000** [comment]
My coworker was confused on what I was asking. Yes we are experiencing the slowness too. The issue is all of the math in https://github.com/igoshev/laravel-captcha/blob/6694a98ed72f58e9c896308cd3fbd29a225937f9/src/Captcha/Generator/GeneratorWaves.php#L68-L136

On individual runs it takes about 0.002 seconds. However once you multiply that by each of the foreachs such as the width (120) and height (80) you get 20 seconds. The smaller the image the less time it would take.

Instead of looking into this further I just switched to: https://github.com/mewebstudio/captcha

Not the ideal solution but it works for now.

**Comment by some0ne2** [comment]
@tm1000 Thanks for update, any idea why this might occur only when we switch to PHP8.1?
Yes I know mewebstudio, but for us this is not ideal too, that's why we switched to this captcha lib.

**Comment by tm1000** [comment]
@some0ne2 it's a bug in php not anything with this library. The math functions are slightly slower in php 8.1 and thus causes slowdowns.

**Comment by tm1000** [comment]
The solution here is probably to replace WaveGenerator with code that uses https://github.com/Intervention/image to generate a similar image (this is what mewebstudio does)

The math however is used to create the wavy text

**Comment by some0ne2** [comment]
Did they replace the wave generator because of the bug in PHP8.1 too?

**Comment by tm1000** [comment]
@some0ne2 they dont use WaveGenerator. WaveGenerator is exclusive to this library

**Comment by some0ne2** [comment]
Someone with free time available? Would provide some cups of coffee (tips) for fixing :).

**Comment by hrace009** [comment]
i think we should upgrade to PHP 8.2

**Comment by some0ne2** [comment]
@hrace009 Not released, anybody an update on to this?
@tm1000 Thanks for verification, is this bug documented somewhere on PHP bugtracker?

**Comment by hrace009** [comment]
@some0ne2 i have seen PHP 8.2 somewhere, but still in dev mode

**Comment by mohammadmahdavim** [comment]
hi every body
Still haven't found a solution for this?

**Comment by some0ne2** [comment]
Any update onto this?
Someone tried 8.2 release of PHP?

**Comment by tm1000** [comment]
I switched captcha providers

**Comment by some0ne2** [comment]
@tm1000 any recommendation?

**Comment by tm1000** [comment]
https://github.com/mewebstudio/captcha
